thoth/user_api/api_v1.py

```diff
@@ -703,12 +703,28 @@ def schedule_kebechet(body: dict):
 
 
 def schedule_kebechet_webhook(body: typing.Dict[str, typing.Any]):
-    """Schedule Kebechet on Openshift."""
+    """Schedule Kebechet run-webhook on Openshift using Argo Workflow."""
+    payload, webhook_payload = {}, {}
+    headers = connexion.request.headers
+    
+    if "X-GitHub-Event" in headers:
+        webhook_payload["event"] = headers["X-GitHub-Event"]
+        webhook_payload["payload"] = body
+        webhook_payload["service"] = "github"
+    elif "X_GitLab_Event" in headers:
+        webhook_payload["payload"] = body
+        webhook_payload["service"] = "gitlab"
+    elif "X_Pagure_Topic" in headers:
+        service = "pagure"
+        return {"error": "Pagure is currently not supported"}, 501
+    else:
+        return {"error": "This webhook is not supported"}, 501
+    
     try:
-        parsed_payload = json.dumps(body)
+        parsed_payload = json.dumps(webhook_payload)
```

Comment:
We do this serialization in the lower level - in OpenShift instance during the actual scheduling to keep the API as high level as possible.

Reply:
Not serializing seems to not allow json.loads to work inside the instance. 
That is when I try to json.loads this variable it's no more a json. 
This is what the env variable looks without serializing - 
https://paste.ofcode.org/uwQBraK2yVu6v5SY79aiYg

Reply:
My point was to move this logic inside the `schedule_kebechet_workflow` so its encapsulated and the method accepts a dict. Similarly as [in this case](https://github.com/thoth-station/common/blob/0790a79873b4227dc2ae1ca9c63e9b63485fec21/thoth/common/openshift.py#L699).

Reply:
Thanks, I have removed the same in https://github.com/thoth-station/user-api/pull/865/commits/1289258cd90fdb2b47bf10979a7f49eca962ee36 :)